service/storage_service.cc

```diff
@@ -6235,12 +6460,25 @@ void storage_service::init_messaging_service(sharded<db::system_distributed_keys
             return ss.cleanup_tablet(tablet);
         });
     });
+    if (raft_topology_change_enabled) {
+        ser::join_node_rpc_verbs::register_join_node_request(&_messaging.local(), [handle_raft_rpc] (raft::server_id dst_id, service::join_node_request_params params) {
+            return handle_raft_rpc(dst_id, [params = std::move(params)] (auto& ss) mutable {
+                return ss.join_node_request_handler(std::move(params));
+            });
+        });
+        ser::join_node_rpc_verbs::register_join_node_response(&_messaging.local(), [handle_raft_rpc] (raft::server_id dst_id, service::join_node_response_params params) {
+            return handle_raft_rpc(dst_id, [params = std::move(params)] (auto& ss) mutable {
+                return ss.join_node_response_handler(std::move(params));
+            });
+        });
+    }
 }
 
 future<> storage_service::uninit_messaging_service() {
     return when_all_succeed(
         _messaging.local().unregister_node_ops_cmd(),
-        ser::storage_service_rpc_verbs::unregister(&_messaging.local())
+        ser::storage_service_rpc_verbs::unregister(&_messaging.local()),
+        ser::join_node_rpc_verbs::unregister(&_messaging.local())
```

Comment:
Did you check that it is fine to unregister if not registered? I hope it is.

Reply:
Yes, I checked and it's safe. It will just be a no-op: https://github.com/scylladb/seastar/blob/576ee47d91fb48bf34801eba182d086b55c28c4f/include/seastar/rpc/rpc_impl.hh#L684